### PR TITLE
Using libmpdel to get mpd information

### PR DESCRIPTION
**Pull request by LucasTavaresA**

### CODE REVIEW

File: awesome-tray.el [hunk 1]
```diff
@@ -703,26 +704,36 @@ These goes before those shown in their full names."
...
-            (progn
-              (setq awesome-tray-mpc-command-last-time current-seconds)
-              (awesome-tray-mpc-command-update-cache))
-          awesome-tray-mpc-command-cache))
+(with-eval-after-load 'libmpdel
+  (add-hook 'libmpdel-current-playlist-changed-hook 'awesome-tray-mpd-command-update-cache)
+  (add-hook 'libmpdel-current-song-changed-hook 'awesome-tray-mpd-command-update-cache))
+
+(defun awesome-tray-module-mpd-info ()
+  (if (executable-find "mpd")
```
Comment (manateelazycat) [inline awesome-tray.el:712]: Use (if (and (ignore-errors (require 'libmpdel)) (executable-find "mpd") ...) instead.

We should use `ignore-errors (require 'xxx)``` check third-part library is installed dynamically.

File: awesome-tray.el [hunk 2]
```diff
@@ -11,13 +11,14 @@
...
 ;; Keywords:
-;; Compatibility: GNU Emacs 27.0.50
+;; Compatibility: GNU Emacs 28.1
 ;;
 ;; Features that might be required by this library:
 ;;
 ;; `cl-lib'
 ;; `subr-x'
 ;; `battery'
+;; `libmpdel'
```
Comment (manateelazycat) [inline awesome-tray.el:21]: Remove this line, awesome-tray shouldn't depend any third-part library

File: awesome-tray.el [hunk 3]
```diff
@@ -206,6 +207,7 @@
 (require 'timer)
 (require 'minibuffer)
 (require 'overlay)
+(require 'libmpdel nil t)
```
Comment (manateelazycat) [inline awesome-tray.el:210]: Remove this line, awesome-tray shouldn't depend any third-part library

**Review by manateelazycat — CHANGES_REQUESTED** [review]

**Comment by LucasTavaresA** [comment]
wait its not working

**Comment by manateelazycat** [comment]
Works now?

**Comment by LucasTavaresA** [comment]
its working fine, but using `(awesome-tray-mode 1)` gives me the error (invalid-read-syntax ) 1 0)

**Comment by manateelazycat** [comment]
toggle-debug-on-error

**Comment by LucasTavaresA** [comment]
[debug log](https://pastebin.com/vu9afJSN)

**Comment by LucasTavaresA** [comment]
im using doom emacs, i used the same config but the problem is in this branch only

**Comment by LucasTavaresA** [comment]
put way to much parenthesis
but its ready now

**Comment by manateelazycat** [comment]
Looks good, thanks for patch.

**Pull request merged**